cfn-resources/custom-db-role/cmd/resource/resource.go

```diff
@@ -49,31 +47,25 @@ func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler
 		return *modelValidation, nil
 	}
 
-	// Create atlas client
-	if currentModel.Profile == nil || *currentModel.Profile == "" {
-		currentModel.Profile = aws.String(profile.DefaultProfile)
-	}
-
-	client, peErr := util.NewMongoDBClient(req, currentModel.Profile)
+	util.SetDefaultProfileIfNotDefined(&currentModel.Profile)
+	client, peErr := util.NewAtlasClient(&req, currentModel.Profile)
 	if peErr != nil {
 		return *peErr, nil
 	}
 
 	atlasCustomDBRole := currentModel.ToCustomDBRole()
-
-	customDBRole, response, err := client.CustomDBRoles.Create(context.Background(), *currentModel.ProjectId, &atlasCustomDBRole)
+	customDBRole, response, err := client.AtlasV2.CustomDatabaseRolesApi.CreateCustomDatabaseRole(context.Background(), *currentModel.ProjectId, atlasCustomDBRole).Execute()
 	if err != nil {
-		if response.Response.StatusCode == http.StatusConflict {
+		if response.StatusCode == http.StatusConflict {
```

Comment:
FYI `apiError, ok := admin.AsError(err)`
```suggestion
		if apiError.Error == http.StatusConflict {
```
and you can ignore the raw http response